server/controllers/RoleController.js

```diff
@@ -0,0 +1,80 @@
+import 'dotenv/config';
+import '@babel/plugin-transform-regenerator';
+import '@babel/polyfill';
+
+import RoleService from '../services/RoleService';
+import Util from '../helpers/util';
+
+const util = new Util();
+
+export default class RoleController {
+  static async create(req, res) {
+    let data = null;
+    // Setup a new Role object
+    const newRole = {
+      name: req.body.name,
+      permissions: req.body.permissions,
+    };
+
+    try {
```

Comment:
Nice job 
What if you just included this into `RoleValidator` into `verifyRoleExist`

Reply:
Since there are some errors that occur directly at the database level, and cannot be caught in the middleware... I opted to use a try-catch block. 

But as @bdushimi mentioned in his [comment](https://github.com/STACK-UP-3/warriorz-backend/pull/19#discussion_r409602664), instead of returning the error response, it's better to send the error to the Winston logs.

But to your point, let me add a check in middleware to see if the Role name already exists.